src/main/java/edu/harvard/iq/dataverse/settings/FileBasedSettingsServiceBean.java

```diff
@@ -0,0 +1,104 @@
+package edu.harvard.iq.dataverse.settings;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+import javax.annotation.PostConstruct;
+import javax.ejb.Singleton;
+import javax.ejb.Startup;
+
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * Service responsible for loading and serving application settings
+ * defined in property files. 
+ * 
+ * @author madryk
+ */
+@Startup
+@Singleton
+public class FileBasedSettingsServiceBean {
+
+	private final static String DEFAULT_PROPERTIES_CLASSPATH_LOCATION = "/config/dataverse.default.properties";
+	private final static String CUSTOM_PROPERTIES_LOCATION = System.getProperty("user.home") + "/.dataverse/dataverse.properties";
+	
+	private static final String BUILTIN_USERS_KEY_SETTING = "BuiltinUsers.KEY";
+	
+	
+	private Map<String, String> settings = new HashMap<>();
+	
+	
+	
+	// -------------------- LOGIC --------------------
```

Comment:
Hmm I'm still not sure about using comments for separating, we can assume that by convention public methods are above private ones. I feel like it's just additional clutter.

Reply:
IMO it's not clutter. It's more like visual separation of concerns in class. If we have clear
division of them - then reading this kind of class is much easier (for me). We have clearly separated "interface" from "internals" and I know at first glance know what is what.